internal/server/commonpool.go

```diff
@@ -0,0 +1,63 @@
+package server
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+)
+
+
+//islandDistribute distributes resources to island as requested 
+func (s *SOMASServer) islandDistribute(resource int, islandID shared.ClientID) { 
+
+	resChan := make(chan clientInfoUpdateResult, 1)
+
+	ci := s.gameState.ClientInfos[islandID]
+	ci.Resources += resource 	
+	resChan <- clientInfoUpdateResult{
+		ID:  islandID,
+		Ci:  ci,
+		Err: nil,
+	}			
+	s.gameState.ClientInfos[islandID] = ci
+	s.gameState.Environment.CommonPool.Resource -= uint(resource)
+}
+
+//islandDeplete depletes island's resource based on the severity of the storm
+//Can resource go to negative?
+func (s *SOMASServer) islandDeplete(porpotionEffect map[shared.ClientID]float64) {
+	nonDeadClients := getNonDeadClientIDs(s.gameState.ClientInfos)
+	N := len(nonDeadClients)
+	resChan := make(chan clientInfoUpdateResult, N)
+	temp := 0.0
+
+	for id, ci := range s.gameState.ClientInfos {
+		temp = float64(ci.Resources)-porpotionEffect[id] 
+		ci.Resources = int(temp)
+			resChan <- clientInfoUpdateResult{
+			ID:  id,
+			Ci:  ci,
+			Err: nil,
+		}
+		s.gameState.ClientInfos[id] = ci
+	}
+
+	close(resChan)
+	for res := range resChan {
+		id, ci := res.ID, res.Ci
+		s.gameState.ClientInfos[id] = ci
+		
+	}
+}
+
+// islandContribute takes resource donation from individual island into the common-pool
+func (s *SOMASServer) islandContribute(resource int, islandID shared.ClientID) {
+	resChan := make(chan clientInfoUpdateResult, 1)
```

Comment:
why channel?

Reply:
I used channels to enable concurrency (something I should not have done, in painful hindsight). Just use normal code?

Reply:
@anhnguyen24 :) ? Haven't looked at go concurrency at all yet myself

Reply:
I honestly have no idea what went through our brains when we redo this part.. the original code works. Probably got a bit bamboozled :p will redo all the code to related to resource in server. Thanks @lhl2617 @JamesTev

Reply:
merry somas

Reply:
![image](https://user-images.githubusercontent.com/50520772/103106082-79be0880-462a-11eb-805c-b0a3165a9914.jpeg)

Reply:
i cri evrytim